### DISCUSSION TITLE
About tmep file(/tmp/fanal-*) capacity

### DISCUSSION THREAD

**Discussion by hojyh1** [aquasecurity/trivy#5461] [Ideas]
### Description

Hi, I'm using your products well.
There's something weird about using it, so I'm reporting it..
When scanning a particular image, I noticed that the file(/tmp/fanal-*) generated by trippy became unusually large in capacity.

I checked several versions(0.45.1, 0.43.1) and it was the same..

```
> trivy -v
Version: 0.45.1
Vulnerability DB:
  Version: 2
  UpdatedAt: 2022-10-05 12:07:40.323319882 +0000 UTC
  NextUpdate: 2022-10-05 18:07:40.323319482 +0000 UTC
  DownloadedAt: 2022-10-05 15:59:49.78634357 +0000 UTC
>
> trivy image --server http://127.0.0.1:4954 --timeout 1800s --format json --output /xxx/xx/tmp/49219209712023102711410546a90232.json --vuln-type os --security-checks vuln,config,secret --ignore-unfixed x.x.x/x-x-x-x/computer-xxxxx-xxxx-4184-88ec-90a17bae8217:latest &
>
> ls -al
total 25692480
.....
-rw-------   1 root     root     24081453056 Oct 21 19:28 fanal-1793837075
-rw-------   1 root     root      9160474624 Oct 21 19:28 fanal-2293919784
-rw-------   1 root     root      6156574720 Oct 21 19:28 fanal-2329256137
-rw-------   1 root     root        24080384 Oct 21 19:29 fanal-2788413697
.....
```

I think it got worse from 0.43.1 version. Can you figure out the cause?
What role is this file?
Is it possible to adjust the size of the file so that it doesn't get too large?

### Target

Container Image

### Scanner

None

**Comment by tillkuhn** [comment]
@hojyh1 I just faced the same issue and our default `/tmp` directory was 100% full during concurrent scans. While it's apparently not possible to control the size, you can point  the env variable`TMPDIR` to a different directory with more capacity. 

Source: https://aquasecurity.github.io/trivy/v0.50/docs/references/troubleshooting/

> Trivy uses the /tmp directory during image scan, if the image is large or /tmp is of insufficient size then the scan fails You can set the TMPDIR environment variable to use redirect trivy to use a directory with adequate storage.
> 
> Try: `$ TMPDIR=/my/custom/path trivy image ...`

**Comment by DmitriyLewen** [comment]
Hello @hojyh1 

Trivy uses fanal-* files to save docker/podman images and large files.
If you scanned large image, you will end up with large `fanal-*` file for that image.

But Trivy deletes these files after the scan is completed.

Regards, Dmitriy
